### DISCUSSION TITLE
Bring back Granite.Avatar

### DISCUSSION THREAD

**Discussion by hanaral** [elementary/granite#470] [General]
I love the simplicity of the Granite library as someone who wants to get in to elementary devving on the side - but replacing a core widget with another library's one is quite offputting if you came to the project because of its unique identity.

At the end of the day 'it's just an avatar widget', but when keeping it seemingly has no negative impact I don't see why it should go

Thanks for coming to my TED talk

**Comment by danirabbit** [comment]
I understand how it would be a little different compared to proprietary platforms, but it is very common on Open Source platforms to use multiple libraries. LibHandy widgets are being used pretty extensively now in elementary apps. Deprecating Granite classes in favor of ones in upstream libraries is nothing new either. Removals are good because we can focus on maintaining code that is being used. Granite Avatar had a number of drawbacks and Hdy.Avatar does exactly what we intended a new avatar widget to do. So instead of duplicating effort, it is better to collaborate

**Reply by hanaral** [reply]
I get that, but it's still a hole that needs to be filled imo - elementary is still a product
Even with styling, some of their stuff looks out of place in elementary (like the lack of fades at the sides of courasel) and the thinness of the hdy.avatar text looks odd too (I presume that this can't be styled?). 
So, I've been thinking about this for a while, how about making granite widgets with hdy when the flatpak ecosystem is up and running?

**Reply by danirabbit** [reply]
If there are improvements you would like to see in Handy, the solution is to file those issues upstream, not to fork
